camel-core/src/main/java/org/apache/camel/component/dataset/DataSetEndpoint.java

```diff
@@ -59,6 +60,8 @@
     private long preloadSize;
     @UriParam(label = "consumer", defaultValue = "1000")
     private long initialDelay = 1000;
+    @UriParam(label = "consumer,producer", defaultValue = "null")
+    private Boolean disableDataSetIndex;
```

Comment:
I really struggled with this - a boolean probably isn't the right type of switch.

What about using values like "strict", "lenient" and "disabled", and the default would be lenient?  Except I'm not sure how you represent that in the UriParam annotation.